### PR TITLE
modified laplacian_matrix to achieve speed-up

### PR DESCRIPTION
**Pull request by dahtah**
For some reason sparse + Diagonal is very slow. Modified laplacian_matrix to use sparse + sparse instead. This achieves a 100x speed-up in some cases.

### CODE REVIEW

**Comment by dahtah** [comment]
Ref:  #1168

**Comment by sbromberger** [comment]
Could you post some benchmarks?

File: src/linalg/spectral.jl [hunk 1]
```diff
@@ -79,8 +80,9 @@ function laplacian_matrix(g::AbstractGraph{U}, T::DataType=Int; dir::Symbol=:uns
         dir = is_directed(g) ? :both : :out
     end
     A = adjacency_matrix(g, T; dir=dir)
-    D = convert(SparseMatrixCSC{T,U}, Diagonal(sparse(sum(A, dims=2)[:])))
-    return D - A
+    s = sum(A;dims=2)
+    D = convert(SparseMatrixCSC{T,U}, spdiagm(0 => s[:]))
+    D-A
```
Comment (sbromberger) [inline src/linalg/spectral.jl]: prefer explicit returns here: `return D - A`.

File: src/linalg/spectral.jl [hunk 2]
```diff
@@ -79,8 +80,9 @@ function laplacian_matrix(g::AbstractGraph{U}, T::DataType=Int; dir::Symbol=:uns
         dir = is_directed(g) ? :both : :out
     end
     A = adjacency_matrix(g, T; dir=dir)
-    D = convert(SparseMatrixCSC{T,U}, Diagonal(sparse(sum(A, dims=2)[:])))
-    return D - A
+    s = sum(A;dims=2)
```
Comment (sbromberger) [inline src/linalg/spectral.jl]: leave a space between arguments and after `;` (I know the previous version didn't do this, but we're making a concerted effort here.)

See L84 as well.

**Comment by dahtah** [comment]
Here are some numbers (v1.0.3 on Linux). On some graphs the gain is massive. For 
```
G = Grid([50,50])
```

The old version gives:

```
BenchmarkTools.Trial: 
  memory estimate:  1.07 MiB
  allocs estimate:  2571
  --------------
  minimum time:     16.163 ms (0.00% GC)
  median time:      16.205 ms (0.00% GC)
  mean time:        16.508 ms (1.55% GC)
  maximum time:     67.344 ms (75.87% GC)
  --------------
  samples:          303
  evals/sample:     1
```
and the new
```
BenchmarkTools.Trial: 
  memory estimate:  914.52 KiB
  allocs estimate:  2560
  --------------
  minimum time:     272.325 μs (0.00% GC)
  median time:      277.983 μs (0.00% GC)
  mean time:        338.724 μs (16.49% GC)
  maximum time:     50.634 ms (99.29% GC)
  --------------
  samples:          10000
  evals/sample:     1
```  

On 
``` 
G = Grid([30,30,30])
```
Old:
```
BenchmarkTools.Trial: 
  memory estimate:  11.96 MiB
  allocs estimate:  27074
  --------------
  minimum time:     1.845 s (0.00% GC)
  median time:      1.877 s (0.00% GC)
  mean time:        1.867 s (0.23% GC)
  maximum time:     1.879 s (0.00% GC)
  --------------
  samples:          3
  evals/sample:     1
```
New:
```
BenchmarkTools.Trial: 
  memory estimate:  11.57 MiB
  allocs estimate:  27060
  --------------
  minimum time:     5.150 ms (0.00% GC)
  median time:      5.881 ms (6.01% GC)
  mean time:        5.727 ms (7.57% GC)
  maximum time:     59.003 ms (89.62% GC)
  --------------
  samples:          872
  evals/sample:     1
```


On other graphs the difference is less spectacular but still significant: 
```
G = SimpleDiGraph(1000,10000)
```
Old:
```
BenchmarkTools.Trial: 
  memory estimate:  2.89 MiB
  allocs estimate:  14782
  --------------
  minimum time:     4.098 ms (0.00% GC)
  median time:      4.164 ms (0.00% GC)
  mean time:        4.418 ms (5.01% GC)
  maximum time:     55.348 ms (92.26% GC)
  --------------
  samples:          1131
  evals/sample:     1
```
New:
```
BenchmarkTools.Trial: 
  memory estimate:  2.87 MiB
  allocs estimate:  14772
  --------------
  minimum time:     1.496 ms (0.00% GC)
  median time:      1.556 ms (0.00% GC)
  mean time:        1.758 ms (10.77% GC)
  maximum time:     52.933 ms (96.66% GC)
  --------------
  samples:          2838
  evals/sample:     1
```

**Comment by sbromberger** [comment]
These benchmarks are really compelling. Thanks.

**Pull request merged**